tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/ScalaJSClassEmitter.scala

```diff
@@ -374,31 +376,25 @@ private[emitter] final class ScalaJSClassEmitter(
             js.ObjectConstr(transformIdent(id) -> js.IntLiteral(0) :: Nil))
     }
 
-    // Options passed to the defineProperty method
-    val descriptor = js.ObjectConstr {
-      // Basic config
-      val base =
-        js.StringLiteral("enumerable") -> js.BooleanLiteral(true) :: Nil
-
-      // Optionally add getter
-      val wget = {
-        if (property.getterBody == EmptyTree) base
-        else {
-          val fun = desugarToFunction(this, className,
-              Nil, property.getterBody, isStat = false)
-          js.StringLiteral("get") -> fun :: base
-        }
-      }
+    // optional getter definition
+    val optGetter = property.getterBody map { body =>
+      val fun = desugarToFunction(this, className, Nil, body, isStat = false)
+      js.StringLiteral("get") -> fun
+    }
 
-      // Optionally add setter
-      if (property.setterBody == EmptyTree) wget
-      else {
-        val fun = desugarToFunction(this, className,
-            property.setterArg :: Nil, property.setterBody, isStat = true)
-        js.StringLiteral("set") -> fun :: wget
-      }
+    // optional setter definition
+    val optSetter = property.setterArgAndBody map { case (arg, body) =>
+      val fun = desugarToFunction(this, className, arg :: Nil,
+          body, isStat = true)
+      js.StringLiteral("set") -> fun
     }
 
+    // Options passed to the defineProperty method
+    val descriptor = js.ObjectConstr(
+      List(js.StringLiteral("enumerable") -> js.BooleanLiteral(true)) ++
+      optGetter ++ optSetter
```

Comment:
I believe this is a bug. ES6 `get()` and `set()` operators define configurable properties ([spec](http://www.ecma-international.org/ecma-262/6.0/#sec-method-definitions-runtime-semantics-propertydefinitionevaluation)), so that's what we should do too. Filed as #2624.